app/javascript/shared/helpers/localStorage.js

```diff
@@ -21,6 +22,24 @@ export const LocalStorage = {
     window.localStorage.setItem(key + ':ts', Date.now());
   },
 
+  setFlag(store, accountId, key, expiry = 24 * 60 * 60 * 1000) {
+    const storeName = store ? `${store}::${accountId}` : store;
+
+    const rawValue = window.localStorage.getItem(storeName);
+    const parsedValue = rawValue ? JSON.parse(rawValue) : {};
+
+    parsedValue[key] = Date.now() + expiry;
+
+    window.localStorage.setItem(storeName, JSON.stringify(parsedValue));
+  },
+
+  getFlag(store, key) {
+    const rawValue = window.localStorage.getItem(store);
```

Comment:
I think you are passing wrong store name here.
```suggestion
    const storeName = store ? `${store}::${accountId}` : store;
    const rawValue = window.localStorage.getItem(storeName);
```